### PR TITLE
Migrate husky v4 to v6

### PR DESCRIPTION
**Pull request by jroebu14**
This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [husky](https://typicode.github.io/husky) ([source](https://togithub.com/typicode/husky)) | [`4.3.8` -> `6.0.0`](https://renovatebot.com/diffs/npm/husky/4.3.8/6.0.0) | [age](https://docs.renovatebot.com/merge-confidence/) | [adoption](https://docs.renovatebot.com/merge-confidence/) | [passing](https://docs.renovatebot.com/merge-confidence/) | [confidence](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>typicode/husky</summary>

### [`v6.0.0`](https://togithub.com/typicode/husky/releases/v6.0.0)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.2.0...v6.0.0)

After being in early access for Open Source projects and Sponsors for a limited time, **I'm happy to announce that husky 6 is MIT again** and can be freely used in commercial projects! :tada:

**Many thanks to the [Open Source projects](https://typicode.github.io/husky/#/?id=used-by) and Companies which have switched to/sponsored the new husky during this period!**

_OSS is my full-time job, please consider sponsoring the development of husky on [GitHub sponsors](https://togithub.com/sponsors/typicode) or [Open Collective](https://opencollective.com/husky). Thank you!_

### Breaking change

-   `husky init` has been moved to its own package (`npx husky-init`)

### Added

-   Programmatically use husky: `require('husky')`
-   TypeScript definitions

### Migrating from husky 4

Husky 6 contains breaking changes. If you're coming from v4, `npm install husky@6` won't be enough.

**Recommended:** see [husky-4-to-6](https://typicode.github.io/husky/#/?id=husky-4-to-6-cli) CLI to automatically migrate your config. There's also a dedicated section in the [docs](https://typicode.github.io/husky/#/?id=migrate-from-v4-to-v6).

**If you're curious why config has changed, you may be interested in reading:**
<https://blog.typicode.com/husky-git-hooks-javascript-config/>

**Also Husky 6 follows official [npm](https://docs.npmjs.com/cli/v7/using-npm/scripts#best-practices) and [Yarn](https://yarnpkg.com/advanced/lifecycle-scripts#a-note-about-postinstall) best practices regarding autoinstall.** It's recommended to use `prepare` script instead (see [usage](https://typicode.github.io/husky/#/?id=usage) in docs).

### [`v5.2.0`](https://togithub.com/typicode/husky/releases/v5.2.0)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.1.3...v5.2.0)

-   Add `set` command to replace hooks (`husky set .husky/pre-commit cmd`)
-   Update `add` command to append command (`husky add .husky/pre-commit cmd`)
-   Improve error messages

### [`v5.1.3`](https://togithub.com/typicode/husky/releases/v5.1.3)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.1.2...v5.1.3)

-   docs: add specific Yarn v2 install/uninstall instructions
-   cli: `husky init` will detect Yarn v2 and initialize accordingly

### [`v5.1.2`](https://togithub.com/typicode/husky/releases/v5.1.2)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.1.1...v5.1.2)

-   docs: recommend `prepare` script instead of `postinstall` ([#&#8203;890](https://togithub.com/typicode/husky/issues/890))
-   cli: `husky init` use `prepare` script ([#&#8203;890](https://togithub.com/typicode/husky/issues/890))

### [`v5.1.1`](https://togithub.com/typicode/husky/releases/v5.1.1)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.1.0...v5.1.1)

-   style(shell): add trailing newlines ([#&#8203;870](https://togithub.com/typicode/husky/issues/870))
-   fix(init): update package.json postinstall

### [`v5.1.0`](https://togithub.com/typicode/husky/releases/v5.1.0)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.9...v5.1.0)

-   Add `husky init`

### [`v5.0.9`](https://togithub.com/typicode/husky/releases/v5.0.9)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.8...v5.0.9)

-   fix(install): do not fail if not inside a Git directory (closes [#&#8203;851](https://togithub.com/typicode/husky/issues/851))

See <https://github.com/typicode/husky/releases/tag/v5.0.0> for v5 release notes

### [`v5.0.8`](https://togithub.com/typicode/husky/releases/v5.0.8)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.7...v5.0.8)

-   fix(install): better handle Git &lt;2.9 ([#&#8203;846](https://togithub.com/typicode/husky/issues/846)) closes [#&#8203;844](https://togithub.com/typicode/husky/issues/844)

### [`v5.0.7`](https://togithub.com/typicode/husky/releases/v5.0.7)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.6...v5.0.7)

-   fix(shell): double quote params ([#&#8203;824](https://togithub.com/typicode/husky/issues/824))

### [`v5.0.6`](https://togithub.com/typicode/husky/releases/v5.0.6)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.5...v5.0.6)

-   Remove unused files

### [`v5.0.5`](https://togithub.com/typicode/husky/releases/v5.0.5)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.4...v5.0.5)

-   Fix custom directory support [#&#8203;827](https://togithub.com/typicode/husky/issues/827)

### [`v5.0.4`](https://togithub.com/typicode/husky/compare/v5.0.3...v5.0.4)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.3...v5.0.4)

### [`v5.0.3`](https://togithub.com/typicode/husky/compare/v5.0.2...v5.0.3)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.2...v5.0.3)

### [`v5.0.2`](https://togithub.com/typicode/husky/compare/v5.0.1...v5.0.2)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.1...v5.0.2)

### [`v5.0.1`](https://togithub.com/typicode/husky/compare/v5.0.0...v5.0.1)

[Compare Source](https://togithub.com/typicode/husky/compare/v5.0.0...v5.0.1)

### [`v5.0.0`](https://togithub.com/typicode/husky/releases/v5.0.0)

[Compare Source](https://togithub.com/typicode/husky/compare/v4.3.8...v5.0.0)

:point_right: See <https://typicode.github.io/husky> for breaking changes and new installation instructions.

#### Note about the license

Husky 5 is released under The Parity Public License.

It doesn't affect you if you're using husky in an Open Source project or if you're a sponsor. You're free to use it as usual and I hope you'll enjoy this new release :heart:

If you're using husky in a commercial project, you may want to consider becoming a sponsor to support the project. You can also try it for 30 days.

**This is only for a limited time, husky will be MIT again later.**

#### Migrating

**Important** Husky v5 brings a lot of improvements but is also very different from v4. Git hooks won't work if you only upgrade husky dependency, existing config needs to be migrated too.

The best way to switch to v5 is to follow the new installation instructions and migrate existing hooks command using `husky add`.

</details>

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +34,11 @@
     "cypress:3rdParty": "yarn cypress -- --project ./3rdPartyCypress/.",
     "cypress:adhoc": "yarn cypress:interactive -- --project ./AdHocCypress/.",
     "dev": "rm -rf build && cp envConfig/local.env .env && run-p webpack:dev:client webpack:dev:server",
+    "license-checker": "apache2-license-checker",
     "lighthouse": "./scripts/lighthouseRun.sh",
+    "line-staged": "lint-staged",
```
Comment (andrewscfc) [inline package.json]:
```suggestion
    "lint-staged": "lint-staged",
```

**Review by andrewscfc — CHANGES_REQUESTED** [review]
Sorry, just a typo that'll stop this working

File: package.json (hunk 1 above)
Comment (jroebu14) [reply package.json]: oops!

**Review by andrewscfc — APPROVED** [review]
ta!

**Comment by jroebu14** [comment]
@andrewscfc turns out we don't need those package scripts. they run fine with yarn in the husky hooks

**Review by ryanmccombe — APPROVED** [review]
Even better than the original 😮

**Pull request merged**